lib/frameworks/jasmine2.js

```diff
@@ -74,11 +74,29 @@ exports.run = function(runner, specs) {
   var reporter = new RunnerReporter(runner);
   jasmine.getEnv().addReporter(reporter);
 
-  // Filter specs to run based on jasmineNodeOpts.grep and jasmineNodeOpts.invert.
-  jasmine.getEnv().specFilter = function(spec) {
+  // Filter specs/suite to run based on jasmineNodeOpts.grep and jasmineNodeOpts.invert.
+  jasmine.getEnv().specFilter = function(spec, suite) {
+    var suiteMatched = isSuiteMatched(suite);
+
+    //If it matched at suite level and invert is true, all the specs will not be included.
+    if(suiteMatched && jasmineNodeOpts.invertGrep){
```

Comment:
This invert logic is just wrong.  If the suite matches but `invertGrep` is `false` then we don't check the spec?  If the suite doesn't match, we don't do any filtering even if `invertGrep` is `false`?